### PR TITLE
feat: Add a new event type:`dismiss`. Add `view_id` and `elapsed` in event body.

### PR DESCRIPTION
**Pull request by icycodes**
Complete TAB-382

### CODE REVIEW

**Comment by wsxiaoys** [comment]
#1148

**Comment by wsxiaoys** [comment]
is this ready for review?

**Review by wsxiaoys — APPROVED** [review]

File: crates/tabby-common/src/api/event.rs [hunk 1]
```diff
@@ -3,14 +3,18 @@ use utoipa::ToSchema;
 
 #[derive(Serialize, Deserialize, ToSchema, Clone, Debug)]
 pub struct LogEventRequest {
-    /// Event type, should be `view` or `select`.
+    /// Event type, should be `view`, `select` or `dismiss`.
```
Comment (boxbeam) [inline crates/tabby-common/src/api/event.rs:6]: Maybe we should use an enum to model this, or `Discriminant<Event>`? I believe I forgot to modify this comment when adding the `Completion` variant - using `Discriminant<Event>` would make it clear through the type system what this is, and not require updating the comment every time a new event is added.

Comment (wsxiaoys) [reply crates/tabby-common/src/api/event.rs:6]: there's no need to modify this, completion event are not logged through this route.

**Review by boxbeam — APPROVED** [review]

**Pull request merged**